### PR TITLE
logging, remove_const before comparison

### PR DESCRIPTION
**Pull request by vmayoral**
This change removes the const value from the logger before
comparing with std::is_same.

### CODE REVIEW

File: rclcpp/resource/logging.hpp.em [hunk 1]
```diff
@@ -91,7 +91,7 @@ def is_supported_feature_combination(feature_combination):
  */
 #define RCLCPP_@(severity)@(suffix)(logger, @(''.join([p + ', ' for p in get_macro_parameters(feature_combination).keys()]))...) \
   static_assert( \
-    ::std::is_same<typename std::remove_reference<decltype(logger)>::type, \
+    ::std::is_same<typename std::remove_reference<std::remove_const<decltype(logger)>::type>::type, \
```
Comment (Karsten1987) [inline rclcpp/resource/logging.hpp.em]: too bad `std::remove_cvref` is not in there yet, but I think `std::remove_cv` is a better replacement for it which also covers volatile.

Comment (wjwwood) [reply rclcpp/resource/logging.hpp.em]: +1 for `std::remove_cv`

Comment (vmayoral) [reply rclcpp/resource/logging.hpp.em]: 284cd94c2c30eca5d0f19d1c7d9a7d14ec24d23b there you go

**Comment by vmayoral** [comment]
Fixed DCO error as well.

**Comment by wjwwood** [comment]
@vmayoral this needs to be rebased as there are now conflicts.

**Comment by vmayoral** [comment]
@wjwwood please review again

**Comment by Karsten1987** [comment]
I went ahead and triggered a CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/6683/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/3002/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/5491/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/6514/)

**Comment by Karsten1987** [comment]
thanks @vmayoral for the patch.

**Comment by wjwwood** [comment]
Yeah, thanks @vmayoral :)

Thanks also @Karsten1987 for pushing on it.

**Pull request merged**